.github/actions/check-branch-match/action.yaml

```diff
@@ -0,0 +1,52 @@
+name: check-branch-match
+description: Check if the current branch matches the specified filters
+
+inputs:
+  branches_include:
+    description: 'Branch names to run this job on, supports wildcards, comma delimited'
+    required: false
+    default: '*'
+  branches_ignore:
+    description: 'Branch names to run this job on, supports wildcards, comma delimited'
+    required: false
+    default: ''
+  ref:
+    description: 'The branch to check'
+    required: false
+    default: ${{ github.ref }}
+
+outputs:
+  match:
+    description: Whether the current branch matches the filters
+    value: ${{ steps.branch_filter.outputs.match }}
+
+runs:
+  using: composite
+  steps:
+    - name: Check for matching branch
+      id: branch_filter
+      uses: actions/github-script@v7
+      with:
+        script: |
+          function wildcardMatch(text, pattern) {
+            const regexPattern =
+            new RegExp('^' + pattern.replace(/\?/g, '.').replace(/\*/g, '.*') + '$');
```

Comment:
is this to simulate glob patterns to match the dorny action?

Reply:
mind adding a short comment about why the regex works the way it does?

Reply:
i originally added this to match how github path filters work